_posts/2020-08-18-scaffolding-static-site-with-cdk.md

```diff
@@ -0,0 +1,243 @@
+---
+layout: post
+title: Scaffolding your AWS static site insfrastructure with CDK
+date: 2020-08-24
+tags: aws cdk static-site react angular cloud-development-kit infrastructure-as-code
+author: Tim Veletta
+---
+
+Many of client jobs we do at Mechanical Rock involve the creation of static sites so it often helps to have a simple, repeatable process for quickly scaffolding one. The AWS Cloud Development Kit (CDK) is a relatively new framework for defining your cloud infrastructure as code to be provisioned with CloudFormation. CDK allows developers of static sites to use the tools they are already familiar with, in this case TypeScript (or JavaScript), to define a template for their infrastructure that can be reused from one project to the next.
```

Comment:
>Many of client jobs we do at Mechanical Rock involve the creation of static sites

This isn't quite true. We do single-page apps, rather than static sites. The rest of this paragraph is great though.

Reply:
True, I think I might be better off using _SPA_ instead of _static site_ throughout!